examples/default/src/App.tsx

```diff
@@ -33,11 +34,15 @@ export const App: React.FC = () => {
     });
   }, []);
 
+  useEffect(() => {
+    Instabug.setNavigationListener(navigationRef);
+  }, [navigationRef]);
+
   return (
     <GestureHandlerRootView style={styles.root}>
       <NativeBaseProvider theme={nativeBaseTheme}>
         <QueryClientProvider client={queryClient}>
-          <NavigationContainer onStateChange={Instabug.onStateChange} theme={navigationTheme}>
+          <NavigationContainer theme={navigationTheme} ref={navigationRef}>
```

Comment:
I think we can keep this one using `onStateChange` and add a new example app with Expo Router using the new API. What do you think?

Reply:
I think adding expo example app is a good idea, but for now the navigation listener covers both cases.
we can change it back to onStateChange afterwards

Reply:
@kholood-ea I think having an Expo app in our examples would still be beneficial for testing other Expo-specific features. Besides it allows us to have both APIs (`Instabug.onStateChange` and `Instabug.setNavigationListener`) in the example apps. What do you think?